assets/app/scripts/directives/nav.js

```diff
@@ -27,7 +27,28 @@ angular.module('openshiftConsole')
       link: function ($scope, element, attrs) {
         var select = $('.selectpicker', element);
 
-        var updateOptions = function(projects) {
+        var updateOptions = function() {
+          var currentProject = $scope.project;
+          if (!currentProject || angular.equals({}, currentProject)) {
```

Comment:
also make sure we have `$scope.projectName` before constructing, just in case that doesn't exist